src/EntropyComputation.cc

```diff
@@ -0,0 +1,107 @@
+
+#include <mcptam/EntropyComputation.h>
+
+
+//this function evaluates a point's entropy reduction
+double EvaluatePoint(Tracker* tracker, MapPoint& point, KeyFrame& tracker_kf, double priorPointCovariance, int pointLevel, double& prevEntropy)
+{
+		
+	//step 1: find the point's anchor keyframe
+	KeyFrame& anchorKF = *point.mpPatchSourceKF;
+	
+	//check to see if the anchorKF and trackerKF are from the SAME MKF parent.  If this is the case, ignore
+	//if (anchorKF.mpParent == tracker_kf.mpParent)
+		//return 0; //todo (adas): do we need this still?
+	
+	//step 2: extract the location of the anchor keyframe
+	TooN::SE3<> anchorCamFromWorld = anchorKF.mse3CamFromWorld;
+	
+	//step 3: compute the relative transform from the anchor keyframe to the current tracker location
+	TooN::SE3<> trackerCamFromWorld = tracker_kf.mse3CamFromWorld;
+	TooN::SE3<> relativeTransform = trackerCamFromWorld*anchorCamFromWorld.inverse(); 
+
+	//step 4: compute the motion derivatives of the point wrt the anchor keyframe
+	
+	 TooN::Vector<3> v3PointInCam = anchorKF.mse3CamFromWorld*point.mv3WorldPos;
+     double dLength = TooN::norm(v3PointInCam);
+     double dRho = 1.0/dLength;
+     TooN::Vector<3> v3PointInCamDir = v3PointInCam * dRho;
+     TooN::Vector<3> v3Axis = v3PointInCamDir ^ TooN::makeVector(0,0,1);
+     double angle = asin(TooN::norm(v3Axis));
+     TooN::normalize(v3Axis);
```

Comment:
I think we need normalization. if v3PointInCamDir=(a,b,c), a^2+b^2+c^2=1 (I think it's the normalized version of v3PointInCam) then this cross product v3PointInCamDir ^ TooN::makeVector(0,0,1) = (b,-a,0) and we know b^2+a^2 is not equal to 1 unless c=0.